SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -500,24 +599,47 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 		return;
 	}
 
-	// If we're the client, we can now ask the server to spawn our controller.
-	if (!IsServer())
+	if (IsServer())
+	{
+		if (GlobalStateManager != nullptr)
+		{
+			// Increment the session id, so users don't rejoin the old game.
+			GlobalStateManager->SetCanBeginPlay(true);
+			GlobalStateManager->TriggerBeginPlay();
+			GlobalStateManager->SetAcceptingPlayers(true);
+			GlobalStateManager->IncrementSessionID();
+		}
+		else
+		{
+			UE_LOG(LogSpatial, Error, TEXT("Map loaded on server but GlobalStateManager is not properly initialised. Session cannot start, players cannot connect."));
+		}
+	}
+	else
 	{
-		// If we know the GSM is already accepting players, simply spawn.
-		if (GlobalStateManager->bAcceptingPlayers && GetWorld()->RemovePIEPrefix(GlobalStateManager->DeploymentMapURL) == GetWorld()->RemovePIEPrefix(GetWorld()->URL.Map))
+		if (CanSendPlayerSpawnRequests())
 		{
-			PlayerSpawner->SendPlayerSpawnRequest();
-			bWaitingForAcceptingPlayersToSpawn = false;
+			MakePlayerSpawnRequest();
 		}
 		else
 		{
-			checkNoEntry();
+			UE_LOG(LogSpatial, Warning, TEXT("Client map finished loading but could not send player spawn request. Will requery the GSM for the correct map to load."));
+			QueryGSMToLoadMap();
 		}
 	}
 
 	bMapLoaded = true;
 }
 
+void USpatialNetDriver::MakePlayerSpawnRequest()
+{
+	if (bWaitingToSpawn)
+	{
+		PlayerSpawner->SendPlayerSpawnRequest();
+		bWaitingToSpawn = false;
+		bPersistSpatialConnection = false;
```

Comment:
Can you add a comment about why this bool is being configured here? On plain reading it's not obvious what a persistent spatial connection has to do with a spawn request.

Reply:
I have no idea why. I mean I understand this will recreate our spatial connection but why do want to do the after making the spawn request? I just kept this from what already existed.